### PR TITLE
enable warnings - first part: warning fixes

### PR DESCRIPTION
**Pull request by furiel**
Proposing to merge those commits that fix some warnings from #1302.  This will reduce the scope from #1302 so that #1302 can only deal with the error/warning reporting. Also these patches contain very good changes that would be great if merged: the ipv6 related warning causes for me a daily headache (--disable-all-modules + CFLAGS=-Werror in configure).

I squashed and split some patches patches to ease the review.

Authors, signoffs are copied over of course. Thank you @bazsi @szemere @bkil-syslogng for your work!

### CODE REVIEW

**Comment by kira-syslogng** [comment]
success

File: modules/stardate/tests/test_stardate.c [hunk 1]
```diff
@@ -28,7 +28,7 @@
 
 #include "cfg.h"
 
-MsgFormatOptions parse_options;
+extern MsgFormatOptions parse_options;
```
Comment (Kokan) [inline modules/stardate/tests/test_stardate.c]: This is actually just a workaround, by not showing the hidden dependency between this test and the `libtest/msg_parse_lib.c`. Due to the `msg_parse_lib` change, all of the unit test that uses `libtest` and the `parse_options` should include the `msg_parse_lib.h` also, and not define/declare `parse_options`.

File: lib/mainloop.c [hunk 2]
```diff
@@ -93,6 +93,7 @@ volatile gint main_loop_workers_running;
 ThreadId main_thread_handle;
 GCond *thread_halt_cond;
 GStaticMutex workers_running_lock = G_STATIC_MUTEX_INIT;
+volatile gint main_loop_workers_running;
```
[the remark this answers is not in the document]
Comment (furiel) [reply lib/mainloop.c]: Very good point! Thanks!

**Comment by kira-syslogng** [comment]
success

**Comment by kira-syslogng** [comment]
success

**Comment by kira-syslogng** [comment]
success

File: lib/template/tests/test_template_compile.c [hunk 3]
```diff
@@ -59,17 +59,28 @@ Plugin hello_plugin = TEMPLATE_FUNCTION_PLUGIN(hello, "hello");
     element.type;  \
     element.msg_ref; }
 
-#define assert_compiled_template(text, default_value, spec, type, msg_ref) \
+#define assert_compiled_template(set_text, set_default_value, spec, type, msg_ref) \
```
[the remark this answers is not in the document]
Comment (Kokan) [reply lib/template/tests/test_template_compile.c]: We have discussed it in IRL, and agreed that it needs some work. But it is not the scope of this PR, and as I said before the change improves the readability of the macro (imho).

**Comment by kira-syslogng** [comment]
success

**Comment by furiel** [comment]
I needed to force push the branch because this commit `*: replace constant putenv() usage with setenv()` conflicted with a recently merged criterion replated pr. While resolving the conflict, I found there were 3 other places where we should have replaced putenv with setenv, so I edited those into the patch. This is the only patch that contains a small new content.

**Comment by Kokan** [comment]
Nice :+1:

**Pull request merged**